src/js/main.js

```diff
@@ -0,0 +1,48 @@
+import Mouse from './mouse';
+import SelectService from './select.service';
+import PolygonService from './polygon.service';
+
+window.onload = () => {
+    let canvas = document.getElementById("canvas"),
+        ctx = canvas.getContext('2d'),
+        mouse = new Mouse(0 ,0),
+        selectService = new SelectService();
+    canvas.setAttribute('height',window.innerHeight - 20);
+    canvas.setAttribute('width',window.innerWidth - 20);
+    let polygons = PolygonService.initializePolygons();
+    PolygonService.drawPolygons(polygons, canvas, ctx);
+
+    canvas.addEventListener('mousedown', (evt) => {
+        mouse.x = evt.clientX;
+        mouse.y = evt.clientY;
+        polygons.some((item) => {
+            if(item.isPointInPolygon(mouse.x, mouse.y, item))
+                selectService.selectedItem = item;
+        });
+    });
+
+    canvas.addEventListener('mousemove', (evt) => {
+        if(selectService.isItemSelected){
+            let indexOfSelectedItem = polygons.indexOf(selectService.selectedItem),
+                item = polygons[indexOfSelectedItem],
+                dx = evt.clientX - mouse.x,
+                dy = evt.clientY - mouse.y;
+            PolygonService.movePolygon(item, dx, dy);
```

Comment:
looks like `movePolygon` should be a method of polygon and looks like `polygon.move()`